### PR TITLE
mutation test first round

### PR DESCRIPTION
**Pull request by Mikearaya**
I've added most of the mutation tests remaining. one thing missing is mutation.publishProduct is not working as expected. I have tried a product with each status but I keep getting `ProductWrongStatusError`  "The current status of the product does not allow this operation".

it works perfectly on the control panel, I don't know what I'm missing. once that figured out it should be ready for a merge. for now, I have slipped it on the test

:bowing_man:

### CODE REVIEW

**Comment by pozylon** [comment]
Oh dude fantastic 🤩

**Comment by Mikearaya** [comment]
> Oh dude fantastic  I can look into this tomorrow I guess I don't have time for it today.

**Comment by Mikearaya** [comment]
> Looks great. I did not go into detail, but the amount of work is impressive. This is really helpful for us!

thank you, I appreciate it. :smile: :bowing_man:

**Comment by pozylon** [comment]
@Mikearaya please rebase to latest master of unchainedshop/unchained

**Comment by Mikearaya** [comment]
@pozylon rebased

File: tests/assortment-product.test.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+          sortKeys: [
+            {
+              assortmentProductId: AssortmentProduct._id,
+              sortKey: 9,
+            },
+          ],
+        },
+      });
+
+      expect(reorderAssortmentProducts[0].sortKey).toEqual(10);
```
[the remark this answers is not in the document]
Comment (Mikearaya) [reply tests/assortment-product.test.js:56]: ok, I'll add that.

File: tests/filters.test.js [hunk 2]
```diff
@@ -283,4 +283,179 @@ describe("Filters", () => {
       });
     });
   });
+
+  describe("mutation.updateFilter for admin User", () => {
```
[the remark this answers is not in the document]
Comment (Mikearaya) [reply tests/filters.test.js:287]: ok, will do.

**Pull request merged**